### PR TITLE
JSON edit widget tree view improvements

### PR DESCRIPTION
**Pull request by domi4484**
## Description

- Added context menu to copy value or key to clipboard
- Values are colored according to type like in text view

image

### CODE REVIEW

File: src/gui/editorwidgets/qgsjsoneditwidget.h [hunk 1]
```diff
@@ -103,13 +108,17 @@ class GUI_EXPORT QgsJsonEditWidget : public QWidget, private Ui::QgsJsonEditWidg
...
+    void refreshTreeViewItemValue( QTreeWidgetItem *treeWidgetItem, const QString &jsonValueString, const QColor &textColor );
 
     QString mJsonText;
 
     FormatJson mFormatJsonMode = FormatJson::Indented;
 
     QStringList mClickableLinkList;
+
+    QAction mCopyValueAction;
+    QAction mCopyKeyAction;
```
Comment (nyalldawson) [inline src/gui/editorwidgets/qgsjsoneditwidget.h:121]: These should be pointers

File: src/gui/editorwidgets/qgsjsoneditwidget.cpp [hunk 2]
```diff
@@ -24,6 +25,8 @@
 
 QgsJsonEditWidget::QgsJsonEditWidget( QWidget *parent )
   : QWidget( parent )
+  , mCopyValueAction( tr( "Copy value" ) )
```
Comment (nyalldawson) [inline src/gui/editorwidgets/qgsjsoneditwidget.cpp:28]:
```suggestion
  , mCopyValueAction( new QAction( tr( "Copy value" ), this ) )
```
etc

**Comment by domi4484** [comment]
Thank you for the review @nyalldawson , I have changed the actions to pointers

**Comment by Saijin-Naib** [comment]
This is so cool!

Would the JSON view be theme-aware in regards to the Code Editor themes?

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**